script.js

```diff
@@ -1,12 +1,225 @@
+    const value = event.target.value.toLowerCase();
+    episodeList = data.filter((episode) => {
+      return (
+        episode.name.toLowerCase().includes(value) ||
+        episode.summary.toLowerCase().includes(value)
+      );
+    });
+    makePageForEpisodes(episodeList); // Renders episodes according to the search input
+  });
+  makePageForEpisodes(episodeList); // Renders all episodes
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  searchStats.innerText = `Displaying ${episodeList.length} / ${totalEpisodes} episodes`; // Shows how many elements are rendered on the page
+  main.innerHTML = ""; // Clears the old list of shows
+  episodeDropdownEl.innerHTML = ""; // Clears the old show's select element
+  main.classList.add("all-episodes");
+
+  // Loops through the episodeList and creates a card type container for each episode
+  episodeList.forEach((episode) => {
+    // The season and episode number with '0' appended to it
+    const season = episode.season.toString().padStart(2, "0");
+    const episodeNo = episode.number.toString().padStart(2, "0");
+
+    const episodeId = episode.name
+      .toLowerCase()
+      .replaceAll(",", "")
+      .replaceAll(" ", "-");
+
+    const episodeOptionEl = document.createElement("option");
+    episodeOptionEl.value = episodeId;
+    episodeOptionEl.innerText = `S${season}E${episodeNo} - ${episode.name}`;
+    episodeDropdownEl.appendChild(episodeOptionEl);
+
+    // Container
+    const episodeContainer = document.createElement("div");
+    episodeContainer.classList.add("episode");
+    episodeContainer.id = episodeId;
+    main.appendChild(episodeContainer);
+
+    const episodeTitleContainer = document.createElement("div");
+    episodeTitleContainer.classList.add("episode-title-container");
+    episodeContainer.appendChild(episodeTitleContainer);
+
+    const episodeTitle = document.createElement("h3");
+    episodeTitle.classList.add("episode-title");
+    episodeTitle.innerText = `${episode.name} - S${season}E${episodeNo}`;
+    episodeTitleContainer.appendChild(episodeTitle);
+
+    const episodeThumbnailContainer = document.createElement("div");
+    episodeThumbnailContainer.classList.add("episode-thumbnail-container");
+    episodeContainer.appendChild(episodeThumbnailContainer);
+
+    const episodeThumbnail = document.createElement("img");
+    episodeThumbnail.classList.add("episode-thumbnail");
+    episodeThumbnail.src = episode.image.medium; // Link to thumbnail image
+    episodeThumbnailContainer.appendChild(episodeThumbnail);
+
+    const episodeDescriptionContainer = document.createElement("div");
+    episodeDescriptionContainer.classList.add("episode-description");
+    episodeDescriptionContainer.innerHTML = episode.summary;
+    episodeContainer.appendChild(episodeDescriptionContainer);
+  });
```

Comment:
Clean and easy to read code with well named variables. Input boxes work well. Good job.

Reply:
Thanks